Improve debugging messages

markstory opened the pull request:
Add which migration is causing the state to be inconsistent.

Refs #533

markstory committed:
Improve debugging messages

Refs #533

othercorey commented on src/TestSuite/Migrator.php:
```diff
@@ -160,7 +160,11 @@ protected function shouldDropTables(Migrations $migrations, array $options): boo
+                Log::write(
+                    'debug',
+                    'One or more migrations need to be applied. ' .
+                    "Migration with source={$migration['name']} id={$migration['id']} needs to be applied."
```
Do we have any way to format a list?

markstory replied:
Not really. We also break out of the loop early. We could instead collect all the output and format a list.

othercorey replied:
Can we at least line break and prefix with something? A `-` or a space or whatever?

markstory committed:
Improve fixture state mis-match logging further.

Make the text more descriptive and broken out into sections. I've also
fixed a few test case related errors I was getting locally related to
caching. I still don't have a clean run on SQLite locally but I'll keep
trying.

markstory committed:
Fix phpcs

othercorey commented:
@dereuromark can you review this?

dereuromark commented on src/TestSuite/Migrator.php:
```diff
@@ -112,7 +112,7 @@ public function runMany(
             if (!$migrations->migrate($migrationSet)) {
                 throw new RuntimeException(
-                    sprintf('Unable to migrate fixtures for `%s`.', $migrationSet['connection'])
+                    "Unable to migrate fixtures for `{$migrationSet['connection']}`."
```
Personally, I like the sprintf() syntax more, in terms of readability etc. but no blocker.
